containers/server-image/salt-event-publisher.service

```diff
@@ -0,0 +1,19 @@
+[Unit]
+Description=Salt Event Publisher
+After=network.target salt-master.service
+Requires=salt-master.service
+
+[Service]
+Type=simple
+User=salt
+StandardOutput=syslog
+StandardError=syslog
+SyslogIdentifier=salt-event-publisher
+ExecStartPre=/bin/bash -c 'until [ -S /run/salt/master/master_event_pub.ipc ]; do sleep 1; done'
+ExecStart=/usr/bin/socat TCP4-LISTEN:4512,reuseaddr,fork UNIX-CONNECT:/run/salt/master/master_event_pub.ipc
```

Comment:
Do we need to expose one more port with this? I'm assuming not, since this port should be for consumption inside the uyuni network and not exposed to the outside world. 
Do you think we need to have some kind of authentication mechanism for it?

Reply:
This basically boils down to sharing the IPC namespace between the saline container and the server one. Did you try https://docs.podman.io/en/latest/markdown/podman-run.1.html#sharing-ipc-between-containers ? In any case this should probably not be the default and only added if saline is enabled.

Reply:
@rjmateus this port is not intended to be exposed outside, but required just for saline container to connect and it's working in one direction only, but exposes the unencrypted data. Do we need to protect it somehow if we are not going to expose it except to the other containers on the server? In case of detaching `salt-master` and `salt-api` we should expose couple more ports the same way to make it possible for `salt-api` to communicate to `salt-master`.

@cbosdo feel bit stupid but I didn't get how it could be applicable here, the section you send the link to is describing interprocess communication with shared memory, but in this case it's just a socket. There is a way to enable `tcp` transport there in the master config, but it's using hardcoded `127.0.0.1` and can't be exposed that way. If I understood the idea wrong, could you please point me to more clear example of using such approach?

Reply:
It's a type of IPC, but not using shared memory, basically it's a file and what I found before that in some cases it's shared as a file volume, I've read the section by the link you pointed, but still didn't get the idea how to make it working. And I can't find any example for such use case.

Reply:
I see the approach proposed by Victor here is more generic, as it won't be relying on `podman` features, and therefore also suitable if we use other container engine.

Reply:
In the kubernetes world Saline would be a sidecar container of the container running Salt master. It seems the common practice there is to share the unix socket in an `emptyDir` volume that is only accessible inside the pod. It would probably be good to follow a similar approach on `podman` : exposing that port using `socat` sounds scary to me.